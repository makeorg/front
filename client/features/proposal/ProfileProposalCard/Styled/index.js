import styled from 'styled-components';
import { CardStyle } from 'Client/ui/Cards';
import { TextColors, BasicColors } from 'Client/app/assets/vars/Colors';

export const ProfileProposalCardStyle = styled(CardStyle)`
  margin: 15px 0;
  &:first-child {
    margin: 0 0 15px;
  }
  &:last-child {
    margin: 15px 0 0;
  }
  &:only-child {
    margin: 0;
  }
  &.proposal-refused,
  &.proposal-postponed,
  &.proposal-pending {
    * {
      color: ${TextColors.MediumGrey};
    }
    header,
    .status-refused {
      color: ${BasicColors.PureWhite};
    }
  }
`;
