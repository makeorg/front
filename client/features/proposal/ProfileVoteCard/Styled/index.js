import styled from 'styled-components';
import { BasicColors, ShadowColors } from 'Client/app/assets/vars/Colors';

export const ProfileVoteCardStyle = styled.div`
  padding: 20px;
  margin-bottom: 20px;
  width: 100%;
  box-shadow: 0 1px 1px 0 #00000080;
  background-color: #0000000c;
`;

export const ProfileHasVotedStyle = styled.div`
  display: inline-flex;
  width: 30px;
  height: 30px;
  justify-content: center;
  align-items: center;
  padding: 5px;
  margin-right: 5px;
  border-radius: 50%;
  border-width: 2px;
  font-size: 12px;
  border-style: solid;
  border-radius: 50%;
  border-color: ${props => props.color};
  overflow: hidden;
  color: ${BasicColors.PureWhite};
  background-color: ${props => props.color};
  box-shadow: 0 1px 1px 0 ${ShadowColors.BlackZeroFiveOpacity};
  svg {
    fill: ${BasicColors.PureWhite};
  }
`;
