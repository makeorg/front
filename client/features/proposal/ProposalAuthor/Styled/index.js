import styled from 'styled-components';
import { TextColors } from 'Client/app/assets/vars/Colors';
import { pxToRem } from 'Shared/helpers/styled';
import { Breakpoints } from 'Client/app/assets/vars/Breakpoints';

export const AuthorInfosStyle = styled.cite`
  font-size: ${pxToRem('14px')};
  color: ${TextColors.MediumGrey};
  font-style: normal;
  overflow: hidden;
  @media (min-width: ${pxToRem(Breakpoints.Desktop)}){
    font-size: ${pxToRem('18px')};
  }
`;
