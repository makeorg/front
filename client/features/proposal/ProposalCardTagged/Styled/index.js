import styled from 'styled-components';
import { Breakpoints } from 'Client/app/assets/vars/Breakpoints';
import { MakeFonts } from 'Client/app/assets/vars/Fonts';
import { intToPx } from 'Shared/helpers/styled';
import { SeparatorStyle } from 'Client/ui/Elements/Separators';

export const ProposalStyle = styled.blockquote`
  max-width: 100%;
  font-size: 12px;
  line-height: normal;
  font-family: ${MakeFonts.RobotoBold};
  font-weight: bold;
  align-self: flex-start;
  margin-top: 10px;
  @media (min-width: ${intToPx(Breakpoints.Desktop)}) {
    font-size: 18px;
  }
`;

export const FooterStyle = styled.div`
  display: flex;
  flex-flow: row;
  align-self: flex-start;
  flex-wrap: wrap;
`;

export const ProposalSeparatorStyle = styled(SeparatorStyle)`
  margin-top: 10px;
`;

export const ProposalCardTaggedStyle = styled.div`
  margin: 20px 0;
`;
