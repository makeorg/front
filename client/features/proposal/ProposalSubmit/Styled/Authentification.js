import styled from 'styled-components';
import { pxToRem } from 'Shared/helpers/styled';

const ProposalSubmitAuthentificationWrapper = styled.div`
  max-width: ${pxToRem('620px')};
  min-height: ${pxToRem('340px')};
  display: flex;
  flex-flow: column;
  justify-content: space-between;
  align-items: center;
  text-align: center;
  margin: ${pxToRem('20px')} auto 0;
`;

export default ProposalSubmitAuthentificationWrapper;
