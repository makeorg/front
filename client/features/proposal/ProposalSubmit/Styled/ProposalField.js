import styled from 'styled-components';
import { pxToRem } from 'Shared/helpers/styled';
import { Breakpoints } from 'Client/app/assets/vars/Breakpoints';
import { NoStyleTextInput } from 'Client/ui/Elements/Form';
import { MiddleRow, FlexElement } from 'Client/ui/Elements/FlexElements';
import { SmallGreyButton, SmallRedButton, IconInButton } from 'Client/ui/Elements/ButtonElements';
import { MakeFonts } from 'Client/app/assets/vars/Fonts';
import { TextColors, ShadowColors } from 'Client/app/assets/vars/Colors';

export const InputWrapper = styled(FlexElement)`
  width: 100%;
`;

export const ButtonWrapper = styled(FlexElement)`
  justify-content: flex-end;
  padding: ${pxToRem('7px')} 0;
  ${props => (props.isFieldExpanded ? 'width: 100%;' : '')};
  @media (min-width: ${pxToRem(Breakpoints.Desktop)}){
    min-width: ${pxToRem('230px')};
    ${props => (props.isFieldExpanded ? 'width: auto;' : '')};
  }
  
`;

export const Label = styled.label`
  font-size: ${pxToRem('14px')};
  line-height: ${pxToRem('40px')};
  margin-right:  ${pxToRem('2.5px')};
  white-space: nowrap;
  @media (min-width: ${pxToRem(Breakpoints.Desktop)}){
    font-size: ${pxToRem('18px')};
    line-height: ${pxToRem('50px')};
    margin-left: ${pxToRem('22px')};
  }
`;

export const Input = styled(NoStyleTextInput)`
  width: 100%;
  max-width: ${pxToRem('780px')};
  font-family: ${MakeFonts.RobotoBold};
  font-weight: bold;
  font-size: ${pxToRem('16px')};
  max-height: ${pxToRem('40px')};
  padding: ${pxToRem('10px')} 0 0 ${pxToRem('5px')};
  resize: none;
  ${props => (props.isFieldExpanded ? `
    height: ${pxToRem('125px')};
    max-height: 100%;`
    : ''
  )};
  @media (min-width: ${pxToRem(Breakpoints.Desktop)}){
    font-size: ${pxToRem('18px')};
    max-height: ${pxToRem('50px')};
    padding: ${pxToRem('15px')} 0 0 ${pxToRem('5px')};
    ${props => (props.isFieldExpanded ? `max-height: ${pxToRem('50px')};` : '')};
  }
`;

export const CharLimit = styled(MiddleRow)`
  font-family: ${MakeFonts.RobotoRegular};
  color: ${TextColors.MediumGrey};
  font-size: ${pxToRem('12px')};
  line-height: ${pxToRem('30px')};
  padding: 0 ${pxToRem('2.5px')};
  @media (min-width: ${pxToRem(Breakpoints.Desktop)}){
    font-size: ${pxToRem('16px')};
    line-height: ${pxToRem('30px')};
    padding: 0 ${pxToRem('16px')};
  }
`;

export const DisabledProposalButton = styled(SmallGreyButton)`
  box-shadow: 0 1px 1px 0 ${ShadowColors.BlackZeroTwoOpacity};
  padding: ${pxToRem('5px')} ${pxToRem('10px')};
`;

export const ProposalButton = styled(SmallRedButton)`
  box-shadow: 0 1px 1px 0 ${ShadowColors.BlackZeroFiveOpacity};
  padding: ${pxToRem('5px')} ${pxToRem('10px')};
`;

export const ProposalIcon = styled(IconInButton)`
  margin: 0;
`;

export const ProposalButtonLabel = styled.span`
  display: inline;
  visibility: visible;
`;
