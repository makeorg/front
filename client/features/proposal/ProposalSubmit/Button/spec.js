import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { ProposalSubmitButtonComponent } from '.';
import { ProposalButtonStyle, DisabledProposalButtonStyle } from '../Styled';

const enabledButtonProps = {
  canSubmit: true,
};
const disabledButtonProps = {
  canSubmit: false,
};
const collapsedField = {
  isFieldExpanded: false,
};
const expandedField = {
  isFieldExpanded: true,
};

describe('ProposalSubmitButtonComponent', () => {
  it('Renders enabled button', () => {
    const wrapper = shallow(
      <ProposalSubmitButtonComponent {...enabledButtonProps} />
    );

    expect(wrapper.find(ProposalButtonStyle)).toHaveLength(1);
    expect(wrapper.find(ProposalButtonStyle).prop('disabled')).toBeUndefined();
  });

  it('Renders disabled button', () => {
    const wrapper = shallow(
      <ProposalSubmitButtonComponent {...disabledButtonProps} />
    );

    expect(wrapper.find(DisabledProposalButtonStyle)).toHaveLength(1);
    expect(wrapper.find(DisabledProposalButtonStyle).prop('disabled')).toBe(
      true
    );
  });

  it('Check a11y rules', () => {
    const wrapper = shallow(<ProposalSubmitButtonComponent />);

    expect(wrapper.find(FontAwesomeIcon).prop('aria-hidden')).toBe(true);
  });
});

describe('ProposalSubmitButtonComponent snapshot', () => {
  it('must match the snapshot by default', () => {
    // TODO need to use shallow with jest for better testing
    const component = renderer
      .create(<ProposalSubmitButtonComponent />)
      .toJSON();
    expect(component).toMatchSnapshot();
  });

  it("must match the snapshot if user can't submit", () => {
    // TODO need to use shallow with jest for better testing
    const component = renderer
      .create(<ProposalSubmitButtonComponent {...disabledButtonProps} />)
      .toJSON();
    expect(component).toMatchSnapshot();
  });

  it('must match the snapshot if user can submit', () => {
    // TODO need to use shallow with jest for better testing
    const component = renderer
      .create(<ProposalSubmitButtonComponent {...enabledButtonProps} />)
      .toJSON();
    expect(component).toMatchSnapshot();
  });

  it('must match the snapshot if field is collapsed', () => {
    // TODO need to use shallow with jest for better testing
    const component = renderer
      .create(<ProposalSubmitButtonComponent {...collapsedField} />)
      .toJSON();
    expect(component).toMatchSnapshot();
  });

  it('must match the snapshot if field is expanded', () => {
    // TODO need to use shallow with jest for better testing
    const component = renderer
      .create(<ProposalSubmitButtonComponent {...expandedField} />)
      .toJSON();
    expect(component).toMatchSnapshot();
  });
});
