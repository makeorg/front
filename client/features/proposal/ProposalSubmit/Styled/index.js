import styled from 'styled-components';
import { pxToRem } from 'Shared/helpers/styled';
import {
  BasicColors,
  BorderColors,
  TextColors,
  ShadowColors,
} from 'Client/app/assets/vars/Colors';
import { Breakpoints, Layouts } from 'Client/app/assets/vars/Breakpoints';
import { NoStyleTextInputStyle } from 'Client/ui/Elements/Form/Styled/Input';
import {
  MiddleRowStyle,
  FlexElementStyle,
} from 'Client/ui/Elements/FlexElements';
import {
  GreyButtonStyle,
  RedButtonStyle,
  IconInButtonStyle,
} from 'Client/ui/Elements/ButtonElements';
import { MakeFonts } from 'Client/app/assets/vars/Fonts';

export const ProposalSubmitAuthentificationWrapperStyle = styled.div`
  max-width: ${pxToRem('620px')};
  min-height: ${pxToRem('340px')};
  display: flex;
  flex-flow: column;
  justify-content: space-between;
  align-items: center;
  text-align: center;
  margin: ${pxToRem('20px')} auto 0;
`;

export const DescriptionWrapperStyle = styled.div`
  margin: ${pxToRem('20px')} auto 0;
`;

export const ProposalSubmitFormStyle = styled.form`
  display: flex;
  justify-content: space-between;
  align-items: flex-start;
  width: 100%;
  padding: 0 ${pxToRem('8px')};
  border: ${pxToRem('1px')} solid ${BorderColors.LightGrey};
  border-radius: ${pxToRem('30px')};
  background-color: ${BasicColors.PureWhite};
  ${props =>
    props.isFieldExpanded
      ? `
    flex-flow: column;
    @media (min-width: ${pxToRem(Breakpoints.Desktop)}){
      flex-flow: row;
    }`
      : ''};
`;

export const ProposalSubmitFormWrapperStyle = styled.aside`
  display: block;
  flex-grow: 0;
  width: 100%;
  max-width: ${pxToRem(Layouts.ContainerWidth)};
  margin: ${pxToRem('10px')} auto 0;
  padding: 0 ${pxToRem('20px')};
  @media (min-width: ${pxToRem(Breakpoints.Desktop)}) {
    margin: ${pxToRem('25px')} auto ${pxToRem('15px')};
  }
`;

export const ProposalInputWrapperStyle = styled(FlexElementStyle)`
  width: 100%;
`;

export const ProposalButtonWrapperStyle = styled(FlexElementStyle)`
  justify-content: flex-end;
  padding: ${pxToRem('7px')} 0;
  ${props => (props.isFieldExpanded ? 'width: 100%;' : '')};
  @media (min-width: ${pxToRem(Breakpoints.Desktop)}) {
    min-width: ${pxToRem('230px')};
    ${props => (props.isFieldExpanded ? 'width: auto;' : '')};
  }
`;

export const ProposalLabelStyle = styled.label`
  font-size: ${pxToRem('14px')};
  line-height: ${pxToRem('40px')};
  margin-right: ${pxToRem('2.5px')};
  white-space: nowrap;
  @media (min-width: ${pxToRem(Breakpoints.Desktop)}) {
    font-size: ${pxToRem('18px')};
    line-height: ${pxToRem('50px')};
    margin-left: ${pxToRem('22px')};
  }
`;

export const ProposalInputStyle = styled(NoStyleTextInputStyle)`
  width: 100%;
  max-width: ${pxToRem('780px')};
  font-family: ${MakeFonts.RobotoBold};
  font-size: ${pxToRem('16px')};
  max-height: ${pxToRem('40px')};
  padding: ${pxToRem('10px')} 0 0 ${pxToRem('5px')};
  resize: none;
  ${props =>
    props.isFieldExpanded
      ? `
    height: ${pxToRem('125px')};
    max-height: 100%;`
      : ''};
  @media (min-width: ${pxToRem(Breakpoints.Desktop)}) {
    font-size: ${pxToRem('18px')};
    max-height: ${pxToRem('50px')};
    padding: ${pxToRem('15px')} 0 0 ${pxToRem('5px')};
    ${props =>
      props.isFieldExpanded ? `max-height: ${pxToRem('50px')};` : ''};
  }
`;

export const ProposalCharLimitStyle = styled(MiddleRowStyle)`
  font-family: ${MakeFonts.RobotoRegular};
  color: ${TextColors.MediumGrey};
  font-size: ${pxToRem('12px')};
  line-height: ${pxToRem('30px')};
  padding: 0 ${pxToRem('2.5px')};
  @media (min-width: ${pxToRem(Breakpoints.Desktop)}) {
    font-size: ${pxToRem('16px')};
    line-height: ${pxToRem('30px')};
    padding: 0 ${pxToRem('16px')};
  }
`;

export const DisabledProposalButtonStyle = styled(GreyButtonStyle)`
  box-shadow: 0 1px 1px 0 ${ShadowColors.BlackZeroTwoOpacity};
  padding: ${pxToRem('5px')} ${pxToRem('10px')};
`;

export const ProposalButtonStyle = styled(RedButtonStyle)`
  box-shadow: 0 1px 1px 0 ${ShadowColors.BlackZeroFiveOpacity};
  padding: ${pxToRem('5px')} ${pxToRem('10px')};
`;

export const ProposalIconStyle = styled(IconInButtonStyle)`
  margin: 0;
`;

export const ProposalButtonLabelStyle = styled.span`
  display: inline;
  visibility: visible;
`;
