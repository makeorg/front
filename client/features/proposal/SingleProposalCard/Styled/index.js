import styled from 'styled-components';
import { pxToRem } from 'Shared/helpers/styled';
import { BasicColors, TextColors } from 'Client/app/assets/vars/Colors';
import { Breakpoints } from 'Client/app/assets/vars/Breakpoints';
import { MakeFonts } from 'Client/app/assets/vars/Fonts';
import {
  MiddleColumnStyle,
  MiddleColumnToRowStyle,
} from 'Client/ui/Elements/FlexElements';
import { SeparatorStyle } from 'Client/ui/Elements/Separators';
import { RedButtonStyle } from 'Client/ui/Elements/ButtonElements';
import { ProposalStyle } from 'Client/features/sequence/Card/Styled';

export const ContentStyle = styled(ProposalStyle)`
  @media (min-width: ${pxToRem(Breakpoints.Desktop)}) {
    font-size: ${pxToRem('30px')};
    line-height: 1.29;
  }
`;

export const InnerProposalStyle = styled(MiddleColumnStyle)`
  width: 100%;
`;

export const FooterStyle = styled.footer`
  width: 100%;
`;

export const FooterContentStyle = styled(MiddleColumnToRowStyle)`
  width: 100%;
`;

export const DescriptionStyle = styled.p`
  width: 100%;
  max-width: ${pxToRem('650px')};
  color: ${TextColors.MediumGrey};
  font-size: ${pxToRem('13px;')};
  > span {
    font-family: ${MakeFonts.RobotoBold};
    color: ${props => props.theme.color};
  }
  @media (min-width: ${pxToRem(Breakpoints.Desktop)}) {
    text-align: right;
    font-size: ${pxToRem('16px;')};
  }
`;

export const FooterContentSeparatorStyle = styled(SeparatorStyle)`
  max-width: ${pxToRem('100px')};
  margin: ${pxToRem('5px')} 0 ${pxToRem('15px')};
  @media (min-width: ${pxToRem(Breakpoints.Desktop)}) {
    max-width: ${pxToRem('2px')};
    height: ${pxToRem('60px')};
    margin: 0 ${pxToRem('25px')};
  }
`;

export const ButtonStyle = styled(RedButtonStyle)`
  text-decoration: none;
  margin: 0 ${pxToRem('5px')};
  &:hover,
  &:focus {
    color: ${BasicColors.PureWhite};
    text-decoration: none;
  }
`;

export const SharingWrapperStyle = styled(MiddleColumnToRowStyle)`
  width: 100%;
  margin: ${pxToRem('15px')} 0;
  @media (min-width: ${pxToRem(Breakpoints.Desktop)}) {
    margin: ${pxToRem('15px')} 0;
  }
`;

export const SharingTitleStyle = styled.h2`
  font-size: ${pxToRem('13px')};
  color: ${TextColors.MediumGrey};
  margin: 0 0 ${pxToRem('10px')};
  @media (min-width: ${pxToRem(Breakpoints.Desktop)}) {
    font-size: ${pxToRem('18px')};
    margin: 0 ${pxToRem('20px')};
  }
`;
