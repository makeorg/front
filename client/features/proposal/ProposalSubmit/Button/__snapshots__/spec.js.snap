// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProposalSubmitButtonComponent snapshot must match the snapshot by default 1`] = `
.c3 {
  display: none;
  visibility: hidden;
}

.c1 {
  margin-right: 0.375rem;
}

.c0 {
  white-space: nowrap;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: baseline;
  -webkit-box-align: baseline;
  -ms-flex-align: baseline;
  align-items: baseline;
  font-family: 'Roboto Condensed Bold',Arial,serif;
  font-size: 0.875rem;
  border: none;
  border-radius: 1.875rem;
  text-transform: uppercase;
  box-shadow: 0 0 4px 0 rgba(0,0,0,0.5);
  padding: 0.3125rem 0.9375rem;
  color: rgb(255,255,255);
  background-color: rgb(118,118,118);
  box-shadow: 0 1px 1px 0 rgba(0,0,0,0.2);
  padding: 0.3125rem 0.625rem;
}

.c2 {
  margin-right: 0.375rem;
  margin: 0;
}

@media (min-width:47.94rem) {
  .c3 {
    display: block;
    visibility: visible;
  }
}

@media (min-width:47.94rem) {
  .c1 {
    margin-right: 0.625rem;
  }
}

@media (min-width:47.94rem) {
  .c0 {
    font-size: 1rem;
    padding: 0.5625rem 1.25rem;
  }
}

@media (min-width:47.94rem) {
  .c2 {
    margin-right: 0.625rem;
  }
}

<button
  className="c0"
  disabled={true}
  id="proposal-submit-button"
  type="submit"
>
  <span
    className="sc-1wm8iti-6 c1 1 c2"
  >
    <svg
      icon="pencil-alt"
      lib="@fortawesome/react-fontawesome"
      prefix="fas"
    />
  </span>
  <div
    className="c3 mvkq91-1 c3"
  >
    common.propose
  </div>
</button>
`;

exports[`ProposalSubmitButtonComponent snapshot must match the snapshot if field is collapsed 1`] = `
.c3 {
  display: none;
  visibility: hidden;
}

.c1 {
  margin-right: 0.375rem;
}

.c0 {
  white-space: nowrap;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: baseline;
  -webkit-box-align: baseline;
  -ms-flex-align: baseline;
  align-items: baseline;
  font-family: 'Roboto Condensed Bold',Arial,serif;
  font-size: 0.875rem;
  border: none;
  border-radius: 1.875rem;
  text-transform: uppercase;
  box-shadow: 0 0 4px 0 rgba(0,0,0,0.5);
  padding: 0.3125rem 0.9375rem;
  color: rgb(255,255,255);
  background-color: rgb(118,118,118);
  box-shadow: 0 1px 1px 0 rgba(0,0,0,0.2);
  padding: 0.3125rem 0.625rem;
}

.c2 {
  margin-right: 0.375rem;
  margin: 0;
}

@media (min-width:47.94rem) {
  .c3 {
    display: block;
    visibility: visible;
  }
}

@media (min-width:47.94rem) {
  .c1 {
    margin-right: 0.625rem;
  }
}

@media (min-width:47.94rem) {
  .c0 {
    font-size: 1rem;
    padding: 0.5625rem 1.25rem;
  }
}

@media (min-width:47.94rem) {
  .c2 {
    margin-right: 0.625rem;
  }
}

<button
  className="c0"
  disabled={true}
  id="proposal-submit-button"
  type="submit"
>
  <span
    className="sc-1wm8iti-6 c1 1 c2"
  >
    <svg
      icon="pencil-alt"
      lib="@fortawesome/react-fontawesome"
      prefix="fas"
    />
  </span>
  <div
    className="c3 mvkq91-1 c3"
  >
    common.propose
  </div>
</button>
`;

exports[`ProposalSubmitButtonComponent snapshot must match the snapshot if field is expanded 1`] = `
.c2 {
  display: none;
  visibility: hidden;
}

.c1 {
  margin-right: 0.375rem;
}

.c0 {
  white-space: nowrap;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: baseline;
  -webkit-box-align: baseline;
  -ms-flex-align: baseline;
  align-items: baseline;
  font-family: 'Roboto Condensed Bold',Arial,serif;
  font-size: 0.875rem;
  border: none;
  border-radius: 1.875rem;
  text-transform: uppercase;
  box-shadow: 0 0 4px 0 rgba(0,0,0,0.5);
  padding: 0.3125rem 0.9375rem;
  color: rgb(255,255,255);
  background-color: rgb(118,118,118);
  box-shadow: 0 1px 1px 0 rgba(0,0,0,0.2);
  padding: 0.3125rem 0.625rem;
}

.c3 {
  display: inline;
  visibility: visible;
}

@media (min-width:47.94rem) {
  .c2 {
    display: block;
    visibility: visible;
  }
}

@media (min-width:47.94rem) {
  .c1 {
    margin-right: 0.625rem;
  }
}

@media (min-width:47.94rem) {
  .c0 {
    font-size: 1rem;
    padding: 0.5625rem 1.25rem;
  }
}

<button
  className="c0"
  disabled={true}
  id="proposal-submit-button"
  type="submit"
>
  <span
    className="c1 sc-1wm8iti-6 c1"
  >
    <svg
      icon="pencil-alt"
      lib="@fortawesome/react-fontawesome"
      prefix="fas"
    />
  </span>
  <span
    className="c2 2 c3"
  >
    common.propose
  </span>
</button>
`;

exports[`ProposalSubmitButtonComponent snapshot must match the snapshot if user can submit 1`] = `
.c3 {
  display: none;
  visibility: hidden;
}

.c1 {
  margin-right: 0.375rem;
}

.c0 {
  white-space: nowrap;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: baseline;
  -webkit-box-align: baseline;
  -ms-flex-align: baseline;
  align-items: baseline;
  font-family: 'Roboto Condensed Bold',Arial,serif;
  font-size: 0.875rem;
  border: none;
  border-radius: 1.875rem;
  text-transform: uppercase;
  box-shadow: 0 0 4px 0 rgba(0,0,0,0.5);
  padding: 0.3125rem 0.9375rem;
  color: rgb(255,255,255);
  background-color: rgb(232,24,68);
  box-shadow: 0 1px 1px 0 rgba(0,0,0,0.5);
  padding: 0.3125rem 0.625rem;
}

.c2 {
  margin-right: 0.375rem;
  margin: 0;
}

@media (min-width:47.94rem) {
  .c3 {
    display: block;
    visibility: visible;
  }
}

@media (min-width:47.94rem) {
  .c1 {
    margin-right: 0.625rem;
  }
}

@media (min-width:47.94rem) {
  .c0 {
    font-size: 1rem;
    padding: 0.5625rem 1.25rem;
  }
}

@media (min-width:47.94rem) {
  .c2 {
    margin-right: 0.625rem;
  }
}

<button
  className="0 c0"
  id="proposal-submit-button"
  tabIndex={0}
  type="submit"
>
  <span
    className="sc-1wm8iti-6 c1 1 c2"
  >
    <svg
      icon="pencil-alt"
      lib="@fortawesome/react-fontawesome"
      prefix="fas"
    />
  </span>
  <div
    className="c3 mvkq91-1 c3"
  >
    common.propose
  </div>
</button>
`;

exports[`ProposalSubmitButtonComponent snapshot must match the snapshot if user can't submit 1`] = `
.c3 {
  display: none;
  visibility: hidden;
}

.c1 {
  margin-right: 0.375rem;
}

.c0 {
  white-space: nowrap;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: baseline;
  -webkit-box-align: baseline;
  -ms-flex-align: baseline;
  align-items: baseline;
  font-family: 'Roboto Condensed Bold',Arial,serif;
  font-size: 0.875rem;
  border: none;
  border-radius: 1.875rem;
  text-transform: uppercase;
  box-shadow: 0 0 4px 0 rgba(0,0,0,0.5);
  padding: 0.3125rem 0.9375rem;
  color: rgb(255,255,255);
  background-color: rgb(118,118,118);
  box-shadow: 0 1px 1px 0 rgba(0,0,0,0.2);
  padding: 0.3125rem 0.625rem;
}

.c2 {
  margin-right: 0.375rem;
  margin: 0;
}

@media (min-width:47.94rem) {
  .c3 {
    display: block;
    visibility: visible;
  }
}

@media (min-width:47.94rem) {
  .c1 {
    margin-right: 0.625rem;
  }
}

@media (min-width:47.94rem) {
  .c0 {
    font-size: 1rem;
    padding: 0.5625rem 1.25rem;
  }
}

@media (min-width:47.94rem) {
  .c2 {
    margin-right: 0.625rem;
  }
}

<button
  className="c0"
  disabled={true}
  id="proposal-submit-button"
  type="submit"
>
  <span
    className="sc-1wm8iti-6 c1 1 c2"
  >
    <svg
      icon="pencil-alt"
      lib="@fortawesome/react-fontawesome"
      prefix="fas"
    />
  </span>
  <div
    className="c3 mvkq91-1 c3"
  >
    common.propose
  </div>
</button>
`;
